File: Angular/src/app/module/product/service/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Proxy } from 'src/app/common/proxy.class';
import { Product } from '../model/product.class';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  productUrl = this.proxyUrl.baseUrl.concat('/products');
  orderUrl = this.proxyUrl.baseUrl.concat('/order');

  constructor(private http: HttpClient, private proxyUrl: Proxy) { }

  /**
   * Get new product
   *
   * @since 02/01/2019
   */
  public getNewProduct() {
    return this.http.get<Product[]>(this.productUrl);
  }

  /**
   * Get order detail
   *
   * @since 01/01/2019
   * @param orderID order ID
   */
  public getOrder(orderID: number, customerID: number) {
    return this.http.get<Product[]>(this.orderUrl + '/detail/' + orderID + "/" + customerID);
  }

  /**
	 * Revenue statistics
	 *
	 * @since  01/02/2019
	 * @param startDate
	 * @param endDate
	 */
  public statistic(startDate: Date, endDate: Date) {
    return this.http.get<Product[]>(this.proxyUrl.baseUrl + '/statistic/' + startDate + '/' + endDate);
  }
}

File: ECommerce/src/main/java/com/satouhikaru/service/ProductService.java
package com.satouhikaru.service;

import com.satouhikaru.entity.Product;

import java.sql.Date;
import java.util.List;

/**
 * @author Pham Nguyen Ha Quang
 * @since  17/11/2018
 */
public interface ProductService {

	/**
	 * Get list of new product
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  02/01/2019
	 *
	 * @return List
	 */
	List<Product> getNewProduct();

	/**
	 * Get order detail
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  01/01/2019
	 *
	 * @param orderID order ID
	 * @param customerID customer ID
	 * @return List
	 */
	List<Product> getOrder(long orderID, long customerID);

	/**
	 * Revenue statistics
	 *
	 * @author Pham Nguyen Ha Quang
	 * @since  01/02/2019
	 *
	 * @param startDate
	 * @param endDate
	 * @return List
	 */
	List<Product> statistic(Date startDate, Date endDate);
	
}